scripts/xbstrap

```diff
@@ -109,53 +109,129 @@ handle_plan_args.parser.add_argument('--keep-going', action='store_true',
 handle_plan_args.parser.add_argument('--progress-file', type=str,
 		help="file that receives machine-ready progress notifications")
 
+# DCC
+def do_list_srcs(args):
+	cfg = xbstrap.base.config_for_dir()
+	plan = xbstrap.base.Plan(cfg)
+	settings = plan.Settings()
+	for src in cfg.all_sources():
+		state = src.check_if_fetched(settings)
+		if (state.missing):
+			print("Unavailable  {}".format(src.name))
+		else:
+			print("Available    {}".format(src.name))
+
+do_list_srcs.parser = main_subparsers.add_parser('list-srcs')
+
+
+# DCC
 def do_fetch(args):
 	cfg = xbstrap.base.config_for_dir()
-	src = cfg.get_source(args.source)
 	plan = xbstrap.base.Plan(cfg)
 	handle_plan_args(cfg, plan, args);
-	plan.wanted.add((xbstrap.base.Action.FETCH_SRC, src))
+
+	if args.all:
+		settings = plan.Settings() # ItemSettings() is not available directly as an initialization... so new member of Plan
+		for src in cfg.all_sources(): # all_sources() is like all_pkgs() except.. it returns Sources... so new member of Config 
+			state = src.check_if_fetched(settings)
```

Comment:
`check_if_fetched` is not necessary here: while omitting it will re-run this step by default, there is the -c command line option to avoid this.

Reply:
The same holds for the other build steps.

Reply:
got it!